### PR TITLE
Modifying  to conform to  bandit standards.

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Modified `assert` statements to conform to `bandit` standards.

### CODE REVIEW

**Review by eslavich — APPROVED** [review]
These seem like reasonable choices to me.  I don't imagine anyone has been catching and handling `AssertionError`, meaning I doubt the change to `ValueError` and `TypeError` will hurt anything.

**Comment by nden** [comment]
This deserves a Change log entry. Start a `1.2.18 (Unreleased)` section.

File: spherical_geometry/graph.py [hunk 1]
```diff
@@ -260,7 +260,8 @@ def _remove_node(self, node):
         ----------
         node : `~Graph.Node` instance
         """
-        assert node in self._nodes
+        if node not in self._nodes:
+            raise AssertionError
```
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply spherical_geometry/graph.py:264]: I removed the check at line 263.

File: spherical_geometry/graph.py [hunk 2]
```diff
@@ -290,8 +291,8 @@ def _add_edge(self, A, B, source_polygons=[]):
         edge : `~Graph.Edge` instance
             The new edge
         """
-        assert A in self._nodes
-        assert B in self._nodes
+        if A not in self._nodes or B not in self._nodes:
+            raise AssertionError
```
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply spherical_geometry/graph.py:295]: I changed all `AssertionError`'s to `ValueError`.

**Review by jdavies-st — APPROVED** [review]

**Review by nden — APPROVED** [review]

**Pull request merged**